### DISCUSSION TITLE
why you use next in your code at 8:01:40  time

### DISCUSSION THREAD

**Discussion by aly202012** [mrdbourke/pytorch-deep-learning#78] [Q&A] [answered]
In time match to 8:01:40 
you use next in your code 

`next(model_1.parameters()).device`

I want to know why you use next.
thank you for your hard work.

**Comment by mrdbourke** [comment, marked as answer]
Hey @aly202012,

`next()` = to take a "next step" in an iterator (see here: https://www.w3schools.com/python/ref_func_next.asp).

I do this to get a single parameter from the model parameters and check its device.

You can try running it without `next()` and see what happens.

**Reply by rkodja** [reply]
anyone still taking this?
i'm at this part and it doesn't appear to be moving to the gpu:
model_1.to(device)
next(model_1.parameters()).device
device(type='cpu')

**Reply by rkodja** [reply]
this worked when i came back from coffee :)
